File: packages/ios/Sources/Components/Button/Button.figma.swift
import SwiftUI
import Figma

struct KozmosButtonConnect: FigmaConnect {
    let component = KozmosButton.self
    let figmaNodeUrl = "https://figma.com/design/Yj4O8p6Y9h2Sa9zJVoAiVY?node-id=77-1055"

    @FigmaString("Label Text")
    var label: String = "Button"

    @FigmaEnum(
        "Variant",
        mapping: [
            "Default": KozmosButtonVariant.`default`,
            "Destructive": KozmosButtonVariant.destructive,
            "Outline": KozmosButtonVariant.outline,
            "Secondary": KozmosButtonVariant.secondary,
            "Ghost": KozmosButtonVariant.ghost,
            "Link": KozmosButtonVariant.link,
            "Glass": KozmosButtonVariant.glass
        ]
    )
    var buttonVariant: KozmosButtonVariant = .default

    @FigmaEnum(
        "Size",
        mapping: [
            "Default": KozmosButtonSize.`default`,
            "Small": KozmosButtonSize.sm,
            "Large": KozmosButtonSize.lg,
            "Icon": KozmosButtonSize.icon
        ]
    )
    var size: KozmosButtonSize = .default

    @FigmaEnum(
        "State",
        mapping: [
            "Default": "default",
            "Disabled": "disabled",
            "Loading": "loading"
        ]
    )
    var state: String = "default"

    var body: some View {
        KozmosButton(
            self.label,
            variant: self.buttonVariant,
            size: self.size,
            isDisabled: self.state == "disabled",
            isLoading: self.state == "loading",
            action: {}
        )
    }
}

File: packages/android/src/main/java/com/kozmos/components/Button/Button.figma.kt
package com.kozmos.components.button

import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import com.figma.code.connect.Figma
import com.figma.code.connect.FigmaConnect
import com.figma.code.connect.FigmaProperty
import com.figma.code.connect.FigmaType

@FigmaConnect("https://figma.com/design/Yj4O8p6Y9h2Sa9zJVoAiVY?node-id=77-1055")
class KozmosButtonConnect {
    @FigmaProperty(FigmaType.Text, "Label Text")
    val label: String = "Button"

    @FigmaProperty(FigmaType.Enum, "Variant")
    val variant: KozmosButtonVariant = Figma.mapping(
        "Default" to KozmosButtonVariant.Default,
        "Destructive" to KozmosButtonVariant.Destructive,
        "Outline" to KozmosButtonVariant.Outline,
        "Secondary" to KozmosButtonVariant.Secondary,
        "Ghost" to KozmosButtonVariant.Ghost,
        "Link" to KozmosButtonVariant.Link,
        "Glass" to KozmosButtonVariant.Glass
    )

    @FigmaProperty(FigmaType.Enum, "Size")
    val size: KozmosButtonSize = Figma.mapping(
        "Default" to KozmosButtonSize.Default,
        "Small" to KozmosButtonSize.Sm,
        "Large" to KozmosButtonSize.Lg,
        "Icon" to KozmosButtonSize.Icon
    )

    @FigmaProperty(FigmaType.Enum, "State")
    val state: String = Figma.mapping(
        "Default" to "default",
        "Disabled" to "disabled",
        "Loading" to "loading"
    )

    @Composable
    fun ComponentExample() {
        KozmosButton(
            onClick = {},
            variant = variant,
            size = size,
            enabled = state != "disabled",
            isLoading = state == "loading"
        ) {
            Text(label)
        }
    }
}
